app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,127 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Models\Post;
+use App\Models\User;
+use App\Repository\PostRepository;
+use Illuminate\Http\Request;
+
+use function Termwind\render;
+
+class PostController extends Controller
+{
+
+    protected $postRepo;
+
+    public function __construct(PostRepository $postRepo){
+        $this->postRepo = $postRepo;
+    }
+
+    /**
+     * Display a listing of the resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function index()
+    {
+        $posts = $this->postRepo->all();
+        return view('posts.index', compact('posts'));
+    }
+    /**
+     * Show the form for creating a new resource.
+     *
+     * @return \Illuminate\Http\Response
+     */
+    public function create()
+    {
+        return view('posts.create');
+    }
+
+    /**
+     * Store a newly created resource in storage.
+     *
+     * @param  \Illuminate\Http\Request  $request
+     * @return \Illuminate\Http\Response
+     */
+    public function store(Request $request)
+    {
+        $request->validate([
+
+            // 'user_id' => auth()->user()->id,
+
+            'title' => 'required',
+            'description' => 'required',
+
+
+        ]);
+        $postRepo = new Post();
```

Comment:
what is the purpose of creating new Post object here. this part also done in repository?